setup/rpi.md

```diff
@@ -0,0 +1,179 @@
+```
+
+Installation
+```
+# make sure your Debian package index is up-to-date
+sudo apt update
+
+# desktop-Full Install: (Recommended) : ROS, rqt, rviz, robot-generic libraries, 2D/3D simulators and 2D/3D perception
+sudo apt install ros-melodic-desktop-full
+```
+
+Dependencies for building packages
+```
+# install 'em
+sudo apt install python-rosdep python-rosinstall python-rosinstall-generator python-wstool build-essential
+
+# initialize rosdep
+sudo rosdep init
+rosdep update
+```
+<!-- todo: pretty sure these aren't needed anymore. Confirm.
+# sensor msgs dependencies
+rosinstall_generator sensor_msgs --rosdistro melodic --deps --wet-only --tar >melodic-sensor_msgs-wet.rosinstall
+wstool init src melodic-sensor_msgs-wet.rosinstall -->
+
+## 3 Setting up ROS environment and building the rover code
+
+### 3.1 Setup ROS build environment
+
+First we'll create a workspace for the rover code. 
+
+On the Raspberry Pi, open up a terminal `(ctl + alt + t)` and then type the following commands:
+
+```
+# Create a catkin workspace directory, which will contain all ROS compilation and 
+# source code files, and move into it
+mkdir -p ~/osr_ws/src && cd ~/osr_ws
+
+# Source your newly created ROS environment
+# EITHER, if you installed ROS melodic
+source /opt/ros/melodic/setup.bash
+# OR, if you installed ROS kinetic
+source /opt/ros/kinetic/setup.bash
+```
+
+### 3.2 Clone and build the rover code
+
+In the newly created catkin workspace you just made, clone this repo:
+```commandline
+cd ~/osr_ws/src
+git clone https://github.com/nasa-jpl/osr-rover-code.git
+
+# install the dependencies
+rosdep install --from-paths src --ignore-src
+catkin_make
+
+# add the generated files to the path so ROS can find them
+source devel/setup.bash
+```
+
+The rover has some customizable settings that will overwrite the default values. Whether you have any changes compared to the defaults or not, you have to manually create these files:
+```
+cd ~/osr_ws/src/osr-rover-code/ROS/osr_bringup/config
+touch physical_properties_mod.yaml roboclaw_params_mod.yaml
+```
+To change any values from the default, modify these files instead so they don't get tracked by git. The files follow the same structure as the default.
+
+
+### 3.3 Add ROS config scripts to .bashrc
+
+The `source...foo.bash` lines above are used to manually configure your ROS environment. We can do this automatically in the future by doing:
+```
+# use "melodic" or "kinetic" below, as appropriate
+echo "source /opt/ros/melodic/setup.bash" >> ~/.bashrc 
+echo "source ~/osr_ws/devel/setup.bash" >> ~/.bashrc
+```
+This adds the `source` lines to `~/.bashrc`, which runs whenever a new shell is opened on the RPi - by logging in via ssh, for example. So, from now on, when you log into the RPi your new command line environment will have the appropriate configuration for ROS and the rover code.
+
+
+## 4 Setting up serial communication on the RPi
```

Comment:
I think there are two steps missing here:

1. remove `console=serial0,115200` from `/boot/firmware/cmdline.txt` ([source: basicmicro](https://resources.basicmicro.com/configuring-the-raspberry-pi-3-serial-port/) also shows that you need to disable that service)
2. enable serial using raspi-config. You can install that using this [script](https://github.com/EmilGus/install_raspi-config/blob/master/install.sh)

Reply:
- The /boot/firmware/cmdline.txt doesn't actually exist in this Ubuntu image. No action to take.
- The `raspi-config` step isn't necessary. I didn't do it.